critiquebrainz/frontend/templates/navbar.html

```diff
@@ -63,3 +64,57 @@
     </div>
   </div>
 </nav>
+
+<script type="text/javascript">
+
+function driverFunction(media_query) {
+  // @media (min-width: @grid-float-breakpoint)
+  if (window.matchMedia("(min-width: 992px)").matches) {
+    searchBox.style.display = 'none';
+    searchBox.setAttribute("onblur", "focusout()");
+    typeSelect.setAttribute("onchange", "select()");
+  }
+  else {
+    searchBox.style.display = 'inline-block';
+    searchBox.removeAttribute("onblur");
+    typeSelect.removeAttribute("onchange");
+  }
+}
+
+function select() {
+  var optionValue = typeSelect.value;
+  typeSelect.style.display = 'none';
+  searchBox.style.display = 'inline-block';
+  searchBox.focus();
+  
+  switch(optionValue){
+    case 'artist': 
+      searchBox.placeholder = "{{ _('Search') }}" + " " + "{{ _('Artist') }}";
+      break;
+    case 'release-group':
+      searchBox.placeholder = "{{ _('Search') }}" + " " + "{{ _('Release group') }}";
+      break;
+    case 'event': 
+      searchBox.placeholder = "{{ _('Search') }}" + " " + "{{ _('Event') }}";
+      break;
+    case 'place':
+      searchBox.placeholder = "{{ _('Search') }}" + " " + "{{ _('Place') }}";
+      break;
+  }
+  
+}
+
+function focusout() {
+     typeSelect.style.display = 'inline-block';
+     searchBox.style.display = 'none';
+}
+
+var searchBox = document.getElementById("navbar-search");
+var typeSelect = document.getElementById("type-selector");
+var mediaQquery = window.matchMedia("(min-width: 992px)");
+// Call listener function at run time
+driverFunction(mediaQquery);    
+// Attach listener function on state changes                        
+mediaQquery.addListener(driverFunction);             
+
+</script>
```

Comment:
The sole reason to include script here is  to use translatable strings for template output.
(Search field placeholder values)